x-pack/dockerlogbeat/main.go

```diff
@@ -0,0 +1,59 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package main
+
+import (
+	"github.com/docker/go-plugins-helpers/sdk"
+	"github.com/elastic/beats/libbeat/common"
+	logpcfg "github.com/elastic/beats/libbeat/logp/configure"
+	_ "github.com/elastic/beats/libbeat/outputs/console"
+	_ "github.com/elastic/beats/libbeat/outputs/elasticsearch"
+	_ "github.com/elastic/beats/libbeat/outputs/fileout"
+	_ "github.com/elastic/beats/libbeat/outputs/logstash"
+	_ "github.com/elastic/beats/libbeat/publisher/queue/memqueue"
+	_ "github.com/elastic/beats/libbeat/publisher/queue/spool"
+	"github.com/elastic/beats/libbeat/service"
+	"github.com/elastic/beats/x-pack/dockerlogbeat/pipelinemanager"
+)
+
+// genNewMonitoringConfig is a hacked-in function to enable a debug stderr logger
+func genNewMonitoringConfig() (*common.Config, error) {
+	cfgObject := make(map[string]string)
+	cfgObject["level"] = "debug"
+	cfgObject["to_stderr"] = "true"
+
+	cfg, err := common.NewConfigFrom(cfgObject)
+	if err != nil {
+		return nil, err
+	}
+	return cfg, nil
+}
+
+func main() {
+	service.BeforeRun()
+	defer service.Cleanup()
+
+	logcfg, err := genNewMonitoringConfig()
+	if err != nil {
+		panic(err)
```

Comment:
why panic? Consider log with Fatal. This will log the message + call os.Exit(1).

Reply:
There's no "nice" way to get logs to users before the HTTP handler starts up, so I figured it didn't make much of a difference.